contrib/enfuse_pro_2.lua

```diff
@@ -0,0 +1,1050 @@
+	"",	-- default
+	entry_widget_tags
+) 
+dt.preferences.register("module_enfuse_pro", "copy_tags",	-- name
+	"bool",	-- type
+	'enfuse_pro: Copy tags from first image by default',	-- label
+	'Changes DEFAULT selection for Copy Tags, Requires restart to take effect.',	-- tooltip
+	true	-- default
+)
+dt.preferences.register("module_enfuse_pro", "exiftool_copy_tags",                -- name
+	"bool",                                                   -- type
+	_('enfuse pro: copy exif data'),                             -- label
+	_('copy the exif tags from the first image to the target'),  -- tooltip
+	true)                                                     -- default                       
+dt.preferences.register("module_enfuse_pro", "add_image_to_db",                   -- name
+	"bool",                                                   -- type
+	_('enfuse pro: add fused image to database'),                -- label
+	_('add the fused image to the darktable database'),          -- tooltip
+	false)                                                    -- default                              
+dt.preferences.register("module_enfuse_pro", "align_use_gpu",                     -- name
+	"bool",                                                   -- type
+	_('enfuse pro: use GPU for remaping'),                       -- label
+	_('set the GPU remapping for image align'),                  -- tooltip
+	false) 
+dt.preferences.register("module_enfuse_pro", "compression_jpeg",   -- name
+	"integer",                                 -- type
+	_('enfuse pro: JPEG compression'),            -- label
+	_('set the compression for JPEG files'),      -- tooltip
+	98,                                        -- default
+	50,                                        -- min
+	100)                                       -- max                     
+dt.preferences.register("module_enfuse_pro", "compression_tiff",   -- name
+	"enum",                                    -- type
+	_('enfuse pro: TIFF compression'),            -- label
+	_('set the compression type for tiff files'), -- tooltip
+	"LZW",                                     -- default
+	"NONE", "DEFLATE","PACKBITS")        -- va                 
+dt.preferences.register("module_enfuse_pro", "image_color_depth",  -- name
+	"enum",                                    -- type
+	_('enfuse pro: image color depth (bit)'),     -- label
+	_('set the output color depth'),              -- tooltip
+	"16",                                      -- default
+	"8","32","r32","r64")                -- values
+
+--AIS bin location
+AIS_path = df.get_executable_path_preference("align_image_stack")
+if not AIS_path then 
+	AIS_path = ""
+end
+AIS_path_widget = dt.new_widget("file_chooser_button"){
+	title = "Select align_image_stack[.exe] file",
+	value = AIS_path,
+	is_directory = false,
+}
+dt.preferences.register("module_enfuse_pro", "ais_path",	-- name
```

Comment:
if you change this to dt.preferences.register("executable_paths", "align_image_stack", then you can get away from the workarounds that you were doing.  This fits in with [get|set]_executable_path_preference, works with check_if_bin_exists, and leverages the other scripts that are storing executable path information.